### PR TITLE
fix/window.gtag is not defined for Next.js-hydration event(#40410)

### LINKED ISSUE #40410: window.gtag is not defined for Next.js-hydration event
### Verify canary release

- [X] I verified that the issue exists in the latest Next.js canary release

### Provide environment information

```sh
    Operating System:
      Platform: linux
      Arch: x64
      Version: Ubuntu 20.04.0 LTS Fri Sep 09 2022 11:36:36 GMT-0700 (Pacific Daylight Time)
    Binaries:
      Node: 16.14.2
      npm: 7.17.0
      Yarn: 1.22.10
      pnpm: 7.9.5
    Relevant packages:
      next: 12.3.1-canary.0
      eslint-config-next: N/A
      react: 17.0.1
      react-dom: 17.0.1
```

### Which example does this report relate to?

with-google-analytics

### What browser are you using? (if relevant)

_No response_

### How are you deploying your application? (if relevant)

_No response_

### Describe the Bug

calling `reportWebVitals` when having google analytics setup using `with-google-analytics` results in `Next.js-hydration` event getting triggered BEFORE gtag is initialized.

This could be problematic in production if someone is just assuming `gtag` is always defined when reporting on web vitals.

### Expected Behavior

When having google analytics setup using the `with-google-analytics` example, I would expect `gtag` to be defined for all events before triggering `reportWebVitals`.
 

### To Reproduce

Here is a link to reproduce: https://stackblitz.com/edit/nextjs-rnaue8?file=pages%2F_app.js,.env.local.example,lib%2Fgtag.js,pages%2Findex.js

### PR DESCRIPTION
**Pull request by dave-hay**
Next.js-hydration now triggered after gtag is initialized. Fixed by adding _document.js and appropriate script elements.
---
## Documentation / Examples

- [X] Related issues linked using fixes #40410
- [X] Make sure the linting passes by running `pnpm lint`
- [X] The "examples guidelines" are followed from [our contributing doc](https://github.com/vercel/next.js/blob/canary/contributing/examples/adding-examples.md)

Next.js-hydration now triggered after gtag is initialized.

### CODE REVIEW

File: examples/with-google-analytics/pages/_document.js [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+        <script
+          dangerouslySetInnerHTML={{
+            __html: `
+            window.dataLayer = window.dataLayer || [];
+            function gtag(){dataLayer.push(arguments);}
+            gtag('js', new Date());
+
+            gtag('config', '${gtag.GA_TRACKING_ID}', {
+              page_path: window.location.pathname,
+            });
```
Comment (SukkaW) [inline examples/with-google-analytics/pages/_document.js:26]:
```suggestion
```

Do not call `config page_path` in the init script.

**Review by SukkaW — CHANGES_REQUESTED** [review]

File: examples/with-google-analytics/pages/_document.js (hunk 1 above)
Comment (hayitsdavid) [reply examples/with-google-analytics/pages/_document.js]: done!

File: examples/with-google-analytics/pages/_document.js [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+      <Head />
+      <body>
+        <Main />
+        <NextScript />
+
+        {/* Global Site Tag (gtag.js) - Google Analytics */}
+        <script
+          async
+          src={`https://www.googletagmanager.com/gtag/js?id=${gtag.GA_TRACKING_ID}`}
+        />
```
Comment (SukkaW) [inline examples/with-google-analytics/pages/_document.js:16]: `https://www.googletagmanager.com/gtag/js` is also loaded in the `_app` aready. We only need to inline `window.dataLayer` and `function gtag(){}` in the `_document`.

```suggestion
```

**Review by SukkaW — CHANGES_REQUESTED** [review]

File: examples/with-google-analytics/pages/_document.js [hunk 3]
```diff
@@ -0,0 +1,26 @@
+import { Html, Head, Main, NextScript } from 'next/document'
+import * as gtag from '../lib/gtag'
```
Comment (SukkaW) [inline examples/with-google-analytics/pages/_document.js:2]:
```suggestion
```

The `gtag` import is also safe to remove.

**Review by SukkaW — CHANGES_REQUESTED** [review]

**Review by ijjk — APPROVED** [review]
Thanks!

**Pull request merged**